dali/operators/image/convolution/gaussian_blur.cc

```diff
@@ -0,0 +1,272 @@
+    axes_count--;
+    DALI_ENFORCE(ImageLayoutInfo::IsChannelLast(layout),
+                 "Only input data with no channels or channel-last is supported.");
+  }
+  bool is_sequence = layout.find('F') >= 0;
+  if (is_sequence) {
+    axes_start++;
+    axes_count--;
+    DALI_ENFORCE(
+        layout.find('F') == 0,
+        make_string("For sequence inputs frames 'F' should be the first dimension, got layout: \"",
+                    layout.str(), "\"."));
+  }
+  DALI_ENFORCE(axes_count <= kMaxDim,
+               make_string("Too many dimensions, found: ", axes_count,
+                           " data axes, maximum supported is: ", kMaxDim, "."));
+  return {axes_start, axes_count, has_channels, is_sequence};
+}
+
+// axes here is dimension of element processed by kernel - in case of sequence it's 1 less than the
+// actual dim
+template <typename T, int axes, bool has_channels>
+class GaussianBlurOpCpu : public OpImplBase<CPUBackend> {
+ public:
+  using Kernel = kernels::SeparableConvolutionCpu<T, T, float, axes, has_channels>;
+  static constexpr int ndim = Kernel::ndim;
+
+  explicit GaussianBlurOpCpu(const OpSpec& spec, const GaussianDimDesc& dim_desc)
+      : spec_(spec), batch_size_(spec.GetArgument<int>("batch_size")), dim_desc_(dim_desc) {}
+
+  bool SetupImpl(std::vector<OutputDesc>& output_desc, const workspace_t<CPUBackend>& ws) override {
+    const auto& input = ws.template InputRef<CPUBackend>(0);
+    int nsamples = input.size();
+    auto nthreads = ws.GetThreadPool().size();
+
+    output_desc.resize(1);
+    output_desc[0].type = input.type();
+    output_desc[0].shape.resize(nsamples, input.shape().sample_dim());
+
+    params_.resize(nsamples);
+    windows_.resize(nsamples);
+
+    kmgr_.template Initialize<Kernel>();
+    kmgr_.template Resize<Kernel>(nthreads, nsamples);
+
+    for (int i = 0; i < nsamples; i++) {
+      params_[i] = GetSampleParams<axes>(i, spec_, ws);
+      windows_[i].PrepareWindows(params_[i]);
+      // We take only last `ndim` siginificant dimensions to handle sequences as well
+      auto elem_shape = input[i].shape().template last<ndim>();
+      auto& req = kmgr_.Setup<Kernel>(i, ctx_, elem_shape, params_[i].window_sizes);
+      // The shape of data stays untouched
+      output_desc[0].shape.set_tensor_shape(i, input[i].shape());
+    }
+    return true;
+  }
+
+  void RunImpl(workspace_t<CPUBackend>& ws) override {
+    const auto& input = ws.template InputRef<CPUBackend>(0);
+    auto& output = ws.template OutputRef<CPUBackend>(0);
+    auto in_shape = input.shape();
+    auto& thread_pool = ws.GetThreadPool();
+
+    for (int i = 0; i < input.shape().num_samples(); i++) {
+      int seq_elements = 1;
+      int64_t stride = 0;
+      if (dim_desc_.is_sequence) {
+        auto shape = input[i].shape();
+        seq_elements = shape[0];
+        stride = volume(shape.begin() + 1, shape.end());
+      }
+      for (int elem_idx = 0; elem_idx < seq_elements; elem_idx++) {
```

Comment:
Can you sort work the the volume?

Reply:
Done, hope it will get replaced with DALI-1473